File: src/config/urls.ts
/**
 * URLS configuration object for ParaBank application
 * 
 * @description This object contains all the main application URLs used throughout
 * the test suite. It provides a centralized location for URL management and
 * ensures consistency across all tests and page objects.
 * 
 * @example
 * ```typescript
 * import { URLS } from '../config/urls';
 * await page.goto(URLS.HOME);
 * ```
 */
export const URLS = {
  BASE_URL: 'https://parabank.parasoft.com',
  HOME: 'https://parabank.parasoft.com/',
  REGISTRATION: 'https://parabank.parasoft.com/parabank/register.htm',
  LOGIN: 'https://parabank.parasoft.com/parabank/index.htm',
  ACCOUNTS_OVERVIEW: 'https://parabank.parasoft.com/parabank/overview.htm',
  NEW_ACCOUNT: 'https://parabank.parasoft.com/parabank/openaccount.htm',
  TRANSFER_FUNDS: 'https://parabank.parasoft.com/parabank/transfer.htm',
  BILL_PAY: 'https://parabank.parasoft.com/parabank/billpay.htm',
  ABOUT_US: 'https://parabank.parasoft.com/parabank/about.htm',
  CUSTOMER_CARE: 'https://parabank.parasoft.com/parabank/contact.htm'
} as const;

/**
 * API_URLS configuration object for ParaBank API endpoints
 * 
 * @description This object contains all the API endpoints used for backend
 * testing and integration. It provides a centralized location for API URL
 * management and ensures consistency across all API tests.
 * 
 * @example
 * ```typescript
 * import { API_URLS } from '../config/urls';
 * const response = await request.get(API_URLS.ACCOUNTS_API);
 * ```
 */
export const API_URLS = {
  BASE_URL: 'https://parabank.parasoft.com/parabank',
  LOGIN_URL: 'https://parabank.parasoft.com/parabank/index.htm',
  ACCOUNTS_API: 'https://parabank.parasoft.com/parabank/services/bank/customers/12212/accounts'
} as const;

File: src/config/urls.py
"""
URL configuration for ParaBank application.
"""
from typing import Dict


class URLs:
    """URL configuration class for ParaBank application."""
    
    BASE_URL = 'https://parabank.parasoft.com'
    HOME = 'https://parabank.parasoft.com/'
    REGISTRATION = 'https://parabank.parasoft.com/parabank/register.htm'
    LOGIN = 'https://parabank.parasoft.com/parabank/index.htm'
    ACCOUNTS_OVERVIEW = 'https://parabank.parasoft.com/parabank/overview.htm'
    NEW_ACCOUNT = 'https://parabank.parasoft.com/parabank/openaccount.htm'
    TRANSFER_FUNDS = 'https://parabank.parasoft.com/parabank/transfer.htm'
    BILL_PAY = 'https://parabank.parasoft.com/parabank/billpay.htm'
    ABOUT_US = 'https://parabank.parasoft.com/parabank/about.htm'
    CUSTOMER_CARE = 'https://parabank.parasoft.com/parabank/contact.htm'


class APIURLs:
    """API URL configuration class for ParaBank API endpoints."""
    
    BASE_URL = 'https://parabank.parasoft.com/parabank'
    LOGIN_URL = 'https://parabank.parasoft.com/parabank/index.htm'
    ACCOUNTS_API = 'https://parabank.parasoft.com/parabank/services/bank/customers/12212/accounts'


# URL dictionaries for easy access
URLS = {
    'BASE_URL': URLs.BASE_URL,
    'HOME': URLs.HOME,
    'REGISTRATION': URLs.REGISTRATION,
    'LOGIN': URLs.LOGIN,
    'ACCOUNTS_OVERVIEW': URLs.ACCOUNTS_OVERVIEW,
    'NEW_ACCOUNT': URLs.NEW_ACCOUNT,
    'TRANSFER_FUNDS': URLs.TRANSFER_FUNDS,
    'BILL_PAY': URLs.BILL_PAY,
    'ABOUT_US': URLs.ABOUT_US,
    'CUSTOMER_CARE': URLs.CUSTOMER_CARE
}

API_URLS = {
    'BASE_URL': APIURLs.BASE_URL,
    'LOGIN_URL': APIURLs.LOGIN_URL,
    'ACCOUNTS_API': APIURLs.ACCOUNTS_API
}
